### PR TITLE
fix: Unit and Integration Tests

### PR DESCRIPTION
**Pull request by puneeter**
- **General test improvements**: Many integration and unit tests were updated to improve error visibility by including result.exception in assertion messages. This helps debug failures by showing exceptions.

- **Support for custom providers**: Test fixtures and test commands were updated to allow specifying a custom Databricks provider (default "azure") via environment variable, replacing hard-coded "azure" values.

- **Handling namespaced pipelines and outputs**:
  - Tests now support and assert the presence of namespaced pipelines (e.g., "namespaced.pipeline") and outputs with dotted names (like output_5.output_5_1).
  - The number of expected tasks in pipeline tests increased (from 5 to 8) to cover these new cases.
  - Checks for task names were updated to accept both simple and namespaced forms.

- **Workflow name sanitization**: In src/kedro_databricks/utils.py, workflow names now replace dots in pipeline names with underscores to ensure valid naming.

- **Databricks config path update**: Changed the KEDRO_LOGGING_CONFIG and related test paths from /Workspace/${workspace.file_path}/... to /${workspace.file_path}/... for consistency and compatibility.

- **Additional test cases**: New test cases were added for package/project names that use dashes and underscores to ensure correct entry point resolution.

- **Pipeline registration and deployment**: Tests and fixtures were updated to register and deploy new namespaced pipelines and outputs, and to check for the appropriate resource files. 

Overall, these changes improve test robustness, add support for more complex pipeline structures, and enhance configuration flexibility.

### CODE REVIEW

File: tests/integration/test_deploy.py [hunk 1]
```diff
@@ -29,6 +29,19 @@ def identity(arg):
             tags=["tag1", "tag2"],
         ),
         node(identity, ["intermediate"], ["output4"], name="node4", tags=["tag2"]),
+        node(
+            identity,
+            ["intermediate"],
+            ["outputs.output_1.output_1_1"],
+            name="outputs.output_1.output_1_1",
+        ),
```
Comment (JenspederM) [inline tests/integration/test_deploy.py:37]: Why are these changes necessary?

File: tests/unit/test_bundle.py [hunk 2]
```diff
@@ -45,13 +45,15 @@ def test_generate_workflow(metadata):
                 ["input", "input1", "input2", "input3", "input4", "input5"],
                 ["output", "output1", "output2", "output3", "output4", "output5"],
             ),
-            "long_identity_input_input1_input2_input3_input4_input5_output_output1_output2_output3_output4_output",  # Last 5 gets truncated
+            # if name not provided, node name is generated based on:
+            # <func_name>__<hash(node_func_mod,node_func_name,node_input,node_output)>
+            "long_identity__faf8b039",
```
Comment (JenspederM) [inline tests/unit/test_bundle.py:50]: Why change these?

File: tests/unit/test_bundle.py [hunk 3]
```diff
@@ -279,6 +281,8 @@ def test_override_resources(jobs, overrides, expected, error):
         ("Fake Project 123 456", "fake-project"),
         ("Fake Project #%", "fake-project"),
         ("# Fake Project #%", "fake-project"),
+        ("my-project-package", "my-project-package"),
```
Comment (JenspederM) [inline tests/unit/test_bundle.py:284]: Why make these?

File: tests/unit/test_bundle.py (hunk 2 above)
Comment (puneeter) [reply tests/unit/test_bundle.py:50]: I think I was testing for latest kedro version where the names are created using hash functions. Reverted this

File: tests/unit/test_bundle.py (hunk 3 above)
Comment (puneeter) [reply tests/unit/test_bundle.py:284]: To test the edge cases. In my case the entrypoint was with `_` which was converted to `-`. So this test checks that it is not changed.

File: tests/integration/test_deploy.py (hunk 1 above)
Comment (puneeter) [reply tests/integration/test_deploy.py:37]: Just to test out the edge cases with names if any.
